burger-react/src/containers/Modal/Modal.js

```diff
@@ -0,0 +1,137 @@
+            this.setState({ finalPrice: this.props.totalPrice });
+        }
+    };
+    inputChange = (event) => {
+        this.setState((prevState) => {
+            return {
+                ...prevState,
+                [event.target.name]: event.target.value,
+            }
+        })
+    }
+    changeCheckbox = (event) => {
+        this.setState((prevState) => {
+            return {
+                ...prevState,
+                checked: !prevState.checked,
+            }
+        })
+    }
+
+    postOrder = (data) => {
+        const body = JSON.stringify({
+            "orderPhone": Number(data.number),
+            "orderAddress": data.address,
+            "orderName": data.name,
+            "orderFast": this.state.checked,
+            "orderProducts": Object.fromEntries(this.props.orderQuantity.map(e => [e.ingredient, e.quantity])),
+            "orderPrice": this.state.finalPrice.toFixed(2),
+        });
+
+        const URLPOST = {
+            method: 'POST',
+            url: 'https://beetroot-burger-app.herokuapp.com/orders',
+            headers: {
+                'Authorization': 'Bearer fbab44e0-5e31-4a93-bc8f-55fe77a066b0',
+                'Content-Type': 'application/json'
+            },
+            data: body,
+        };
+
+        axios(URLPOST)
+            .then(res => {
+                console.log(res)
+                this.props.setMessage(false)
+            })
+            .catch(error => {
+                console.log(error)
+                this.props.setMessage(true)
+            });
+        this.props.cleanState();
+        this.setState({
+            name: "",
+            number: "",
+            address: "",
+            checked: false,
+            promo: "hello",
+            finalPrice: 1,
+        });
+        this.props.getNew();
+        // this.props.onRequestClose();
+
+    }
+
+    render() {
+        return (
+            <Modal
+                style={customStyles}
+                isOpen={this.props.isOpen}
+                onRequestClose={this.props.onShowHideModal}
+                orderQuantity={this.props.orderQuantity}
+                ariaHideApp={false}
+            >
+                <div className="modal">
+                    <h2 className="modal-title">Your order</h2>
+                    <div>{this.props.orderQuantity.map((e, i) => e.quantity > 0 ? (<p className="order" key={i}>{e.ingredient} x {e.quantity}</p>) : '')}</div>
+                    <h3 className="modal-price">Burger price: {this.state.finalPrice.toFixed(2)} ₴</h3>
+                    <Form
+                        inputChange={this.inputChange}
+                        changeCheckbox={this.changeCheckbox}
+                        checkPromo={this.checkPromo}
+                        postOrder={this.postOrder}
+                        onShowHideModal={this.props.onShowHideModal}
+                        modalState={this.state} />
+                </div>
+            </Modal>
+        )
+    }
+
+}
+
```

Comment:
Modal should be a component. The only mission for Modal - is to render data and trigger functions from props. The parent Container should handle all logic